opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -77,8 +77,14 @@ impl Tracer {
     ) -> Option<(u8, Vec<KeyValue>, TraceState)> {
         let provider = self.provider()?;
         let sampler = &provider.config().default_sampler;
+
+        let ctx = parent_context.map(|span_context| {
+            let span = Span::new(span_context.clone(), None, self.clone());
+            Context::new().with_span(span)
```

Comment:
If you add another parameter to this private method to pass through the `cx: &Context` from the caller, then at least the correct other context variables should be present. E.g. `tracer.build_with_context(builder, some_non_current_context)`

```suggestion
            cx.with_span(span)
```